### PR TITLE
BWS-81: * fix bugs

### PR DESCRIPTION
**Pull request by Oleg-feriss**

### CODE REVIEW

File: backend/src/services/worker/worker.service.ts [hunk 1]
```diff
@@ -233,13 +242,13 @@ class Worker {
...
         const id = item.id;
-        return this.#spaceService.delete({ id, token });
+        return this.#spaceService.delete({ id });
       }),
     );
     await Promise.all(
       slcFunctions.map((slcFunction) => {
         const id = slcFunction.id;
-        return this.#slcFunctionService.delete({ id, token });
+        return this.#slcFunctionService.delete({ id });
```
Comment (Yaroslaww-1) [inline backend/src/services/worker/worker.service.ts]: I don't like this solution, honestly. It is not obvious why token in SlcFunctionService.delete can be null. You have 2 possible options:
1. Do not use slcFunctionService.delete and spaceService.delete. And just copy code from these methods into WorkerService.deleteWorker. (can't recommend this approach)
2. Move role checks to the api layer (as with permissions check)

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: backend/src/api/slc/slc.api.ts [hunk 2]
```diff
@@ -118,11 +127,16 @@ const initSLCApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
...
-        id: req.params.id,
-        token,
-      });
+      if (userRole !== UserRole.WORKER) {
+        throw new SLCError({
+          status: HttpCode.DENIED,
+          message: ExceptionMessage.MASTER_FUNCTION_DELETE,
+        });
+      }
+
```
Comment (what1s1ove) [inline backend/src/api/slc/slc.api.ts:155]:
```suggestion
```
we can't do it here

File: backend/src/api/bs/bs.api.ts [hunk 3]
```diff
@@ -86,11 +96,16 @@ const initBsApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
...
-      await spaceService.delete({
-        id,
-        token,
-      });
+      if (user.userRole !== UserRole.WORKER) {
+        throw new BsError({
+          status: HttpCode.DENIED,
+          message: ExceptionMessage.MASTER_SPACE_DELETE,
+        });
+      }
```
Comment (what1s1ove) [inline backend/src/api/bs/bs.api.ts]:
```suggestion
```
we can't do it here

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/api/bs/bs.api.ts (hunk 3 above)
Comment (Yaroslaww-1) [reply backend/src/api/bs/bs.api.ts]: We can actually

File: backend/src/services/worker/worker.service.ts [hunk 4]
```diff
@@ -214,6 +214,15 @@ class Worker {
       throw new EAMError();
     }
```
Comment (Yaroslaww-1) [inline backend/src/services/worker/worker.service.ts]: Move it to api layer as you done with slc/bs delete methods

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Review by Yaroslaww-1 — APPROVED** [review]

**Pull request merged**